agent/api/task/service_connect_validation.go

```diff
@@ -0,0 +1,293 @@
+			}
+			// Save the listener port value
+			portsMap[listenerPortValue] = true
+		}
+	}
+
+	return nil
+}
+
+// validateBridgeIngressConfig validates the service connect ingress config for bridge network mode.
+func validateBridgeIngressConfig(scIngressConfigList []IngressConfigEntry) error {
+	hostPortsMap := map[uint16]bool{}
+	hostPortValue := uint16(0)
+	listenerPortsMap := map[uint16]bool{}
+	listenerPortValue := uint16(0)
+
+	for _, entry := range scIngressConfigList {
+		// show a warnning messgae if an intercept port exists in the ingress config for bridge mode
+		if entry.InterceptPort != nil {
+			logger.Warn("Service connect config validation: an intercept port should not exist for bridge mode", logger.Fields{
+				"listenerName":  entry.ListenerName,
+				"listenerPort":  entry.ListenerPort,
+				"hostPort":      aws.Uint16Value(entry.HostPort),
+				"interceptPort": aws.Uint16Value(entry.InterceptPort),
+			})
+		}
+
+		// verify the host port
+		if entry.HostPort != nil {
+			hostPortValue = aws.Uint16Value(entry.HostPort)
+			if err := validatePort(hostPortValue); err != nil {
+				return fmt.Errorf("the host port=%d in the ingress config entry is not valid: %w", hostPortValue, err)
+			}
+
+			if present := hostPortsMap[hostPortValue]; present {
+				return fmt.Errorf("host port collision detected in the ingress config entry with the host port=%d, and listener name=%s", hostPortValue, entry.ListenerName)
+			}
+			// Save the host port value
+			hostPortsMap[hostPortValue] = true
+		}
+
+		// verify the listener port
+		if entry.ListenerPort > uint16(0) {
+			listenerPortValue = entry.ListenerPort
+			if err := validatePort(listenerPortValue); err != nil {
+				return fmt.Errorf("the listener port=%d in the ingress config entry is not valid: %w", listenerPortValue, err)
+			}
+
+			if present := listenerPortsMap[listenerPortValue]; present {
+				return fmt.Errorf("listener port collision detected in the ingress config entry with the listener port=%d, and listener name=%s", listenerPortValue, entry.ListenerName)
+			}
+			// Save the listener port value
+			listenerPortsMap[listenerPortValue] = true
+		}
+	}
+
+	return nil
+}
+
+//validatePort validates port is in valid range.
+func validatePort(port uint16) error {
+	// valid port range is 1~65535
+	if port >= uint16(1) && port <= uint16(65535) {
+		return nil
+	}
+
+	return fmt.Errorf("The port=%d is an invalid port. A valid port ranges from 1 through 65535", port)
+}
+
+// ValidateSCConfig validates service connect container name, config, egress config, and ingress config.
+func ValidateServiceConnectConfig(scConfig *ServiceConnectConfig, taskContainers []*ecsacs.Container, taskNetworkMode string, ipv6Enabled bool) error {
+	// container name is a required field
+	if err := validateContainerName(scConfig.ContainerName, taskContainers); err != nil {
+		return err
+	}
+
+	// egress config is a field that can be empty
```

Comment:
Can we also check that egress and ingress config are not both nil?

Reply:
```
// egress config and ingress config should not both be nil/empty
	if scConfig.EgressConfig == nil && len(scConfig.IngressConfig) == 0 {
		return fmt.Errorf("egress config and ingress config should not both be nil/empty")
	}
```
is added to agent/api/task/service_connect_validator.go in the commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look. Thanks!